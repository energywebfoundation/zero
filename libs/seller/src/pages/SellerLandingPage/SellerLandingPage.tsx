import {
  CallToActionButton,
} from '@energyweb/zero-ui-core';
import { Box, Typography } from '@material-ui/core';
import { Helmet } from 'react-helmet-async';
import { useTranslation } from 'react-i18next';
import { useSellerLandingPageEffects } from './SellerLandingPage.effects';

export const SellerLandingPage = () => {
  const {
    handlers: { navigateToSignupPageHandler },
  } = useSellerLandingPageEffects();
  const { t } = useTranslation();

  return (
    <>
      <Helmet>
        <title>Seller</title>
      </Helmet>
      <Box width={'100%'} textAlign={'center'}>
        <Box my={'40px'}>
          <Typography
            mb={'24px'}
            fontSize={'24px'}
            variant={'h5'}
            lineHeight={'31px'}
            fontWeight={600}
          >
            {t('sellerLandingPage.whatIsZero')}
          </Typography>
          <Typography
            lineHeight={'51px'}
            variant={'h3'}
            fontSize={'40px'}
            fontWeight={'700'}
            color={'secondary'}
          >
            {t('sellerLandingPage.zeroIsGlobalSearchEngineForRenewableEnergy')}
          </Typography>
        </Box>
        <Box my={'40px'}>
          <CallToActionButton
            onClick={navigateToSignupPageHandler}
            translateKey={'sellerLandingPage.callToAction'}
          />
        </Box>
      </Box>
    </>
  );
};
