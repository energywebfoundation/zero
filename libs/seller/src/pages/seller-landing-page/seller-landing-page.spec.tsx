import { render } from '@testing-library/react';

import SellerLandingPage from './seller-landing-page';
import { StoreProvider } from '@energy-web-zero/store-configure';
import { UiTheme } from '@energyweb/zero-theme';

describe('SellerLandingPage', () => {
  it('should render successfully', () => {
    const { baseElement } = render(
      <UiTheme>
        <StoreProvider>
          <SellerLandingPage />
        </StoreProvider>
      </UiTheme>
    );
    expect(baseElement).toBeTruthy();
  });

  it('should match snap', () => {
    const { baseElement } = render(
      <UiTheme>
        <StoreProvider>
          <SellerLandingPage />
        </StoreProvider>
      </UiTheme>
    );
    expect(baseElement).toMatchSnapshot();
  });
});
