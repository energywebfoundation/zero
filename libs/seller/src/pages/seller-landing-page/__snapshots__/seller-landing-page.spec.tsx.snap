// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SellerLandingPage should match snap 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-fzwid6"
    >
      <div
        class="MuiBox-root css-rm9e7r"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 css-1v540lc-MuiTypography-root"
        >
          sellerLandingPage.whatIsZero
        </h5>
        <h3
          class="MuiTypography-root MuiTypography-h3 css-106db7z-MuiTypography-root"
        >
          sellerLandingPage.zeroIsGlobalSearchEngineForRenewableEnergy
        </h3>
      </div>
      <div
        class="MuiBox-root css-rm9e7r"
      >
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root makeStyles-root-3 css-1d4ibc5-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          sellerLandingPage.callToAction
          <span
            class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeMedium makeStyles-icon-4 css-1frrczp-MuiSvgIcon-root"
              data-testid="PersonAddAltOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 9V6h-2v3h-3v2h3v3h2v-3h3V9h-3zM9 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0-6c1.1 0 2 .9 2 2s-.9 2-2 2-2-.9-2-2 .9-2 2-2zm6.39 8.56C13.71 13.7 11.53 13 9 13s-4.71.7-6.39 1.56C1.61 15.07 1 16.1 1 17.22V20h16v-2.78c0-1.12-.61-2.15-1.61-2.66zM15 18H3v-.78c0-.38.2-.72.52-.88C4.71 15.73 6.63 15 9 15c2.37 0 4.29.73 5.48 1.34.32.16.52.5.52.88V18z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>
</body>
`;
